jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/WebSocketLocalEndpoint.java

```diff
@@ -28,25 +28,24 @@
  */
 public interface WebSocketLocalEndpoint
 {
-    Logger getLog();
+    Logger getLog(); // TODO why?
+    boolean isOpen(); // TODO Does the endpoint really know this?
 
-    boolean isOpen();
-    
-    void onOpen();
+    default void onOpen() {}
 
-    void onClose(CloseStatus close);
+    default void onClose(CloseStatus close) {}
 
-    void onFrame(Frame frame);
+    default void onFrame(Frame frame) {}
 
-    void onError(Throwable cause);
+    default void onError(Throwable cause) {}
 
-    void onText(Frame frame, Callback callback);
+    default void onText(Frame frame, Callback callback) {}
```

Comment:
I disagree with defaults, this is not an API for users.
This is a core level API for other APIs to build off of, each of those APIs needs to implement each of these interfaces so they understand what is required of them.

Reply:
OK.